plugins/goframe/net/ghttp/intercepter.go

```diff
@@ -0,0 +1,68 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ghttp
+
+import (
+	"fmt"
+	"github.com/apache/skywalking-go/plugins/core/operator"
+	"github.com/apache/skywalking-go/plugins/core/tracing"
+	"net/http"
+)
+
+// GoFrameServerInterceptor is used to intercept and trace HTTP requests.
+type GoFrameServerInterceptor struct{}
+
+// BeforeInvoke intercepts the HTTP request before invoking the handler.
+func (h *GoFrameServerInterceptor) BeforeInvoke(invocation operator.Invocation) error {
+	request := invocation.Args()[1].(*http.Request)
+	s, err := tracing.CreateEntrySpan(fmt.Sprintf("%s:%s", request.Method, request.URL.Path), func(headerKey string) (string, error) {
+		return request.Header.Get(headerKey), nil
+	}, tracing.WithLayer(tracing.SpanLayerHTTP),
+		tracing.WithTag(tracing.TagHTTPMethod, request.Method),
+		tracing.WithTag(tracing.TagURL, request.Host+request.URL.Path),
+		tracing.WithComponent(5004))
```

Comment:
this framework is new, please add a [new library in the skywalking project](https://github.com/apache/skywalking/blob/master/oap-server/server-starter/src/main/resources/component-libraries.yml)

Reply:
how to add a  library in the skywalking project？ 
PR 2 this project?

Reply:
Please first submit a new PR into `component-libraries.yml` in skywalking project as I mentioned above. 
Then, please update the component ID in this file.

Reply:
```suggestion
		tracing.WithComponent(5022))
```

Please update the component ID with the newer ID.